framework/src/positions/FunctorExtremaPositions.C

```diff
@@ -112,45 +117,41 @@ FunctorExtremaPositions::initialize()
           // fallthrough
         case ExtremeType::MAX:
         {
-          if (value > extrema[i])
+          if (value > *extremum)
             extrema_found = true;
           break;
         }
         case ExtremeType::MIN:
         {
-          if (value < extrema[i])
+          if (value < *extremum)
             extrema_found = true;
           break;
         }
       }
       if (extrema_found)
       {
         // Move the extrema down the list of extrema
-        // TODO: optimization could be to use a std::list with a constant insertion time here
-        Real temp = value;
-        Point temp_loc = elem->true_centroid();
-        for (const auto j : make_range(i, _n_extrema))
-        {
-          const auto other_temp = extrema[j];
-          const auto other_temp_loc = extrema_locs[j];
-          extrema[j] = temp;
-          extrema_locs[j] = temp_loc;
-          temp = other_temp;
-          temp_loc = other_temp_loc;
-        }
+        extrema.insert(extremum, value);
+        extrema_locs.insert(extremum_loc, elem->true_centroid());
+        extrema.pop_back();
+        extrema_locs.pop_back();
         // Found an extremum, filled the extrema vector, done
         break;
       }
+      // Consider the next extremum
+      std::advance(extremum, 1);
+      std::advance(extremum_loc, 1);
```

Comment:
beautiful code indeed. Could have also used `std::next` but I was the one who brought up `std::advance`